admin-console/src/main/webapp/src/pages/ApplicationPage/useApplicationTypes.ts

```diff
@@ -0,0 +1,78 @@
+import produce, { original } from 'immer';
+import { useEffect, useState } from 'react';
+import { useClient } from 'react-fetching-library';
+
+import {
+    Operations
+} from '../../generated/OpenapiInternal';
+import { useApplicationTypes as useGetApplicationTypes } from '../../services/EventTypes/GetApplication';
+import {AggregationCardRow, AggregationTemplate } from '../../types/Notifications';
+
+interface ApplicationController {
+    payload: any;
+    data: ReadonlyArray<AggregationCardRow> | undefined;
+    loading: boolean;
+    reload: () => void;
+    error: any;
+}
+
+export const useApplicationTypes = (applicationId: string): ApplicationController => {
+    const [ aggregationRows, setAggregationRows ] = useState<ReadonlyArray<AggregationCardRow>>();
+    const applicationTypesQuery = useGetApplicationTypes(applicationId);
+
+    const { query } = useClient();
+
+    useEffect(() => {
+        setAggregationRows(undefined);
+    }, [ applicationId ]);
+
+    useEffect(() => {
+        if (applicationTypesQuery.payload?.status === 200) {
+            const application = applicationTypesQuery.payload.value.map<AggregationCardRow>(e => ({
```

Comment:
application here is showing that it is cast to any `const application: any` but it needs to show AggregationCardRow, could you take a look by chance?

Reply:
It's that type apparently (`Application1`) but you can't interate over it with a map, because it's a single element.
It should be:
```
const application = {
                ...applicationTypesQuery.payload.value,
                aggregationEmail: {
                    isLoading: true
                }
            };
```